pkg/local/preflight/checker.go

```diff
@@ -490,31 +533,91 @@ func (local *Checker) checkNFSv4Support() error {
 // status and logs a warning if there are fewer than 2 ready replicas.
 //
 // https://github.com/longhorn/longhorn/issues/9752
-func (local *Checker) checkKubeDNS() {
+func (local *Checker) checkKubeDNS() error {
 	logrus.Info("Checking if CoreDNS has multiple replicas")
+	const topic = consts.PreflightCheckTopicKubeDNS
 
 	deployments, err := commonkube.ListDeployments(local.kubeClient, metav1.NamespaceSystem, map[string]string{consts.KubeAppLabel: consts.KubeAppValueDNS})
 	if err != nil {
-		local.collection.Log.Error = append(local.collection.Log.Error, fmt.Sprintf("Failed to list Kube DNS with label %s=%s: %v", consts.KubeAppLabel, consts.KubeAppValueDNS, err))
-		return
+		return wrapErrorWithTopic(topic,
+			fmt.Errorf("failed to list Kube DNS with label %s=%s: %v",
+				consts.KubeAppLabel, consts.KubeAppValueDNS, err))
+
 	}
 
 	if len(deployments.Items) != 1 {
-		local.collection.Log.Warn = append(local.collection.Log.Warn, fmt.Sprintf("Found %d deployments with label %s=%s; expected 1", len(deployments.Items), consts.KubeAppLabel, consts.KubeAppValueDNS))
-		return
+		local.collection.Log.Warn = append(local.collection.Log.Warn,
+			wrapMsgWithTopic(topic, fmt.Sprintf(
+				"found %d deployments with label %s=%s; expected exactly 1",
+				len(deployments.Items), consts.KubeAppLabel, consts.KubeAppValueDNS)))
+		return nil
 	}
 
 	deployment := deployments.Items[0]
 
 	if deployment.Spec.Replicas == nil || *deployment.Spec.Replicas < 2 {
-		local.collection.Log.Warn = append(local.collection.Log.Warn, fmt.Sprintf("Kube DNS %q is set with fewer than 2 replicas; consider increasing replica count for high availability", deployment.Name))
-		return
+		local.collection.Log.Warn = append(local.collection.Log.Warn,
+			wrapMsgWithTopic(topic, fmt.Sprintf("Kube DNS %q is set with fewer than 2 replicas; consider increasing replica count for high availability", deployment.Name)))
+		return nil
 	}
 
 	if deployment.Status.ReadyReplicas < 2 {
-		local.collection.Log.Warn = append(local.collection.Log.Warn, fmt.Sprintf("Kube DNS %q has fewer than 2 ready replicas; some replicas may not be running or ready", deployment.Name))
-		return
+		local.collection.Log.Warn = append(local.collection.Log.Warn,
+			wrapMsgWithTopic(topic, fmt.Sprintf("Kube DNS %q has fewer than 2 ready replicas; some replicas may not be running or ready", deployment.Name)))
+		return nil
+	}
+
+	local.collection.Log.Info = append(local.collection.Log.Info,
+		wrapMsgWithTopic(topic,
+			fmt.Sprintf("Kube DNS %q is set with %d replicas and %d ready replicas", deployment.Name, *deployment.Spec.Replicas, deployment.Status.ReadyReplicas)))
+
+	return nil
+}
+
+func wrapMsgWithTopic(topic, msg string) string {
+	return fmt.Sprintf("%s %s", topic, msg)
+}
+
+func wrapErrorWithTopic(topic string, err error) error {
+	return fmt.Errorf("%s: %w", topic, err)
+}
+
+func wrapMultiErrors(topic, msg string, items map[string]interface{}) error {
```

Comment:
Don't need a new wrapper error since we eventually conclude these items, a `map[string]error`, into a string to send back to the remote collector. It wraps the input into the report.

```suggestion
func wrapMultiErrors(topic, msg string, items map[string]error) string {
```

Reply:
I need this function to return an error because it's used to collect and propagate errors from check functions.